### ISSUE TITLE
NuGet.org - which branch is it built from?

### ISSUE THREAD

**Issue by colin-young** [serilog/serilog#679]
If I'm grabbing the latest beta from https://www.nuget.org/packages/Serilog/2.0.0-beta-502, last updated 22 Feb 2016, is that the latest code from `dev` or is it from `master`?

The reason I'm asking is that when I pull down 502, I'm not seeing the new LogContext support for Core CLR. Or at least my test harness code is breaking when I switch from my private build of the `dev` branch to the nuget.org package. But master appears to be 3 months old, while the package has been updated several times in the last few weeks.

**Comment by merbla** [comment]
Hey @colin-young,
All beta packages should from `dev` ATM.

**Comment by colin-young** [comment]
Thanks. That's what I was assuming. Now I just need to figure out why my PR isn't showing up even though the changes look like they are in there.

**Comment by merbla** [comment]
Is this the PR? 

https://ci.appveyor.com/project/serilog/serilog/build/487

**Comment by colin-young** [comment]
Yup. That's the one. When I say "not seeing" I mean when I replace my private package with the package from nuget.org I get an exception when trying to use LogContext. I didn't want to spend too much time digging into it if there was some other obvious explanation. Sounds like there isn't so I'll need to figure it out.

**Comment by merbla** [comment]
Might also be an AppVeyor build issue.  Let me know how you go.

**Comment by colin-young** [comment]
This may be a beta versioning issue. I had versioned my private builds as `2.0.0-ps-beta-2` and now I'm trying to replace those with `2.0.0-beta-502` which is, as far as NuGet is concerned, older than mine. However, I've removed them from my private NuGet server, cleared the cache, manually deleted any folders on disk referring to my versions, but I still keep finding in my local packages folder a `Serilog.2.0.0-ps-beta-2` folder. I have no idea where that is coming from.

I think something is getting confused somewhere about which version of the package is being used, although either version should have the `LogContext` code in it.

**Comment by colin-young** [comment]
I sorted out my issues with the bad references coming back (that was due to Visual Studio 2015), but I'm still seeing the same problem. When I use DotPeek to disassemble the cached assembly from the dotnet5.4 folder, I see Serilog.Context and within that I see a LogContext type, both of which the build is complaining about.

I'm now starting to worry that my PR was bad and I somehow tested the wrong thing...

**Comment by nblumhardt** [comment]
Hi Colin; if I `Install-Package Serilog -Pre` into a fresh _Console Application (Package)_ project, `LogContext` is there and usable in both the .NET Core and .NET 4.5.1 builds.

Do you have the latest tools installed? You may need to `dnvm upgrade`....?

**Comment by colin-young** [comment]
Nope. All up to date. I see the same thing as you see with the fresh console app. So I'm back to the problem being something on my end. I've had a fair number of issues with nuget packages of the wrong version restoring themselves from who-knows-where, and nothing I've done to clear and delete caches has been able to solve the problem.

That's really how I started and just wanted to confirm that it was in fact me, and not an out of date build. And, I'm glad to know that I didn't completely stuff up the implementation.

Thanks for the help @merbla and @nblumhardt. I'll update this thread once I sort out what's going on.

**Comment by colin-young** [comment]
I wonder if this might have something to do with it:

      "Serilog.Sinks.Seq/2.0.0-beta-34": {
        "type": "package",
        "dependencies": {
          "Serilog": "2.0.0-beta-465",
          "Serilog.Sinks.PeriodicBatching": "2.0.0-beta-465",
          "Serilog.Sinks.RollingFile": "2.0.0-beta-465"
        },
        "frameworkAssemblies": [
          "Microsoft.CSharp",
          "mscorlib",
          "System",
          "System.Core",
          "System.Net.Http"
        ],
        "compile": {
          "lib/net45/Serilog.Sinks.Seq.dll": {}
        },
        "runtime": {
          "lib/net45/Serilog.Sinks.Seq.dll": {}
        }

That appeared to eventually pull in a 465 build of Serilog which does not have the necessary changes in it. I added explicit references to `Serilog.Sinks.PeriodicBatching` and `Serilog.Sinks.RollingFile` version `2.0.0-beta-505` for both, and that seems to have solved things locally (for now). I've been having a lot of trouble with NuGet resurrecting old beta versions that no longer exist anywhere that I can find them (including a full search of the disk, and they've been deleted from the NuGet feed), yet it somehow brings them back, and then I was having trouble with getting `Serilog-2.0.0-beta-505` in one project and `Serilog-2.0.0-beta-502` in another within the same solution, and I ended up going through the `csproj` of yet a different project in the solution to update the hint paths, and that finally fixed that issue.

I'm pretty satisfied that this is a tooling issue and not a Serilog issue (although the Seq sink might be contributing and it would be nice to get a build that references the 505 build of Serilog).
